src/server/frontend_wayland/layer_shell_v1.cpp

```diff
@@ -359,9 +359,8 @@ auto mf::LayerSurfaceV1::get_anchored_edge() const -> MirPlacementGravity
         return mir_placement_gravity_center;
 }
 
-auto mf::LayerSurfaceV1::get_exclusive_rect() const -> std::optional<geom::Rectangle>
+auto mf::LayerSurfaceV1::get_exclusive_rect(int zone) const -> std::optional<geom::Rectangle>
 {
-    auto zone = exclusive_zone.pending();
```

Comment:
It is not meaningful to pass `zone` to `LayerSurfaceV1::get_exclusive_rect()` as a surface only has one, optional, exclusive rect to get! Essentially, this introduces the unnecessary precondition that the one and only valid value is passed. (Either the change is wrong, or the function name is wrong (c.f. `compute_exclusive_rect_for(zone)`)